packages/ai/src/ui/chat.test.ts

```diff
@@ -102,7 +102,11 @@ describe('Chat', () => {
         transport: new DefaultChatTransport({
           api: 'http://localhost:3000/api/chat',
         }),
-        onFinish: () => finishPromise.resolve(),
+        onFinish: ({ message, messages }) => {
+          expect(messages).toBe(chat.messages);
+          expect(message).toBe(chat.lastMessage);
```

Comment:
expect in callback is very hard to debug if it fails. i recommend storing in variables and using expect in the main test code